src/native/managed/cdacreader/Microsoft.Diagnostics.DataContractReader.Contracts/Contracts/CodeVersions_1.cs

```diff
@@ -16,6 +18,85 @@ public CodeVersions_1(Target target)
         _target = target;
     }
 
+    ILCodeVersionHandle ICodeVersions.GetActiveILCodeVersion(TargetPointer methodDesc)
+    {
+        // CodeVersionManager::GetActiveILCodeVersion
+        IRuntimeTypeSystem rts = _target.Contracts.RuntimeTypeSystem;
+        MethodDescHandle md = rts.GetMethodDescHandle(methodDesc);
+        TargetPointer mtAddr = rts.GetMethodTable(md);
+        TypeHandle typeHandle = rts.GetTypeHandle(mtAddr);
+        TargetPointer module = rts.GetModule(typeHandle);
+        uint methodDefToken = rts.GetMethodToken(md);
+        return FindActiveILCodeVersion(module, methodDefToken);
+    }
+
+    ILCodeVersionHandle ICodeVersions.GetILCodeVersion(NativeCodeVersionHandle nativeCodeVersionHandle)
+    {
+        // NativeCodeVersion::GetILCodeVersion
+        IRuntimeTypeSystem rts = _target.Contracts.RuntimeTypeSystem;
+        if (!nativeCodeVersionHandle.Valid)
+        {
+            return ILCodeVersionHandle.Invalid;
+        }
+
+        if (!IsExplicit(nativeCodeVersionHandle))
+        {
+            // There is only a single synthetic NativeCodeVersion per
+            // method and it must be on the synthetic ILCodeVersion
+            MethodDescHandle md = rts.GetMethodDescHandle(nativeCodeVersionHandle.MethodDescAddress);
+            TargetPointer mtAddr = rts.GetMethodTable(md);
+            TypeHandle typeHandle = rts.GetTypeHandle(mtAddr);
+            TargetPointer module = rts.GetModule(typeHandle);
+            uint methodDefToken = rts.GetMethodToken(md);
+            return new ILCodeVersionHandle(module, methodDefToken, TargetPointer.Null);
+        }
+        else
+        {
+            // Otherwise filter all the ILCodeVersions for the one that matches the version id
+            NativeCodeVersionNode nativeCodeVersionNode = AsNode(nativeCodeVersionHandle);
+            foreach (ILCodeVersionHandle ilCodeVersionHandle in ((ICodeVersions)this).GetILCodeVersions(nativeCodeVersionNode.MethodDesc))
+            {
+                if (GetId(ilCodeVersionHandle) == nativeCodeVersionNode.ILVersionId)
+                {
+                    return ilCodeVersionHandle;
+                }
+            }
+        }
+
+        return ILCodeVersionHandle.Invalid;
+    }
+
+    IEnumerable<ILCodeVersionHandle> ICodeVersions.GetILCodeVersions(TargetPointer methodDesc)
+    {
+        // CodeVersionManager::GetILCodeVersions
+        IRuntimeTypeSystem rts = _target.Contracts.RuntimeTypeSystem;
+        MethodDescHandle md = rts.GetMethodDescHandle(methodDesc);
+        TargetPointer mtAddr = rts.GetMethodTable(md);
+        TypeHandle typeHandle = rts.GetTypeHandle(mtAddr);
+        TargetPointer module = rts.GetModule(typeHandle);
+        uint methodDefToken = rts.GetMethodToken(md);
```

Comment:
I see these lines repeated 3-4 times, could we have a `GetModuleAndMethodDef` helper?

Reply:
Added helper